content/academy/puppeteer_playwright/common_use_cases/logging_into_a_website.md

```diff
@@ -0,0 +1,396 @@
+
+await page.close();
+
+// Email sending logic
+const promises = emailsToSend.map(({ to, subject, body }) =>
+    (async () => {
+        const sendEmailContext = await browser.newContext();
+        await sendEmailContext.addCookies(cookies);
+        const page2 = await sendEmailContext.newPage();
+
+        await page2.goto('https://mail.yahoo.com/');
+
+        await page2.click('a[aria-label="Compose"]');
+
+        await page2.type('input#message-to-field', to);
+        await page2.type('input[data-test-id="compose-subject"]', subject);
+        await page2.type('div[data-test-id="compose-editor-container"] div[contenteditable="true"]', body);
+
+        await page2.click('button[title="Send this email"]');
+
+        await sendEmailContext.close();
+    })()
+);
+
+await Promise.all(promises);
+
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteer" lang="javascript">
+import puppeteer from 'puppeteer';
+
+const emailsToSend = [
+    {
+        to: 'abcdef123randomFakeEmail@gmail.com',
+        subject: 'Hello',
+        body: 'This is a message.',
+    },
+    {
+        to: 'testingtesting12345903@aol.com',
+        subject: 'Testing',
+        body: 'I love the academy!',
+    },
+    {
+        to: 'jimmyJohnBillyBob420@academy.net',
+        subject: 'Apify is awesome!',
+        body: 'Some content.',
+    },
+];
+
+const browser = await puppeteer.launch({ headless: false });
+const page = await browser.newPage();
+
+// Login logic
+await page.goto('https://www.yahoo.com/');
+
+await Promise.all([page.waitForSelector('a[data-ylk*="sign-in"]'), page.click('button[name="agree"]')]);
+await Promise.all([page.waitForNavigation(), page.click('a[data-ylk*="sign-in"]')]);
+
+await page.type('input[name="username"]', 'academy_playwright_login');
+await Promise.all([page.waitForNavigation(), page.click('input[name="signin"]')]);
+
+await page.type('input[name="password"]', 'AcademyIsGreat88');
+await Promise.all([page.waitForNavigation(), page.click('button[name="verifyPassword"]')]);
+
+const cookies = await page.cookies();
+await page.close();
+
+// Email sending logic
+const promises = emailsToSend.map(({ to, subject, body }) =>
+    (async () => {
+        const sendEmailContext = await browser.createIncognitoBrowserContext();
+        const page2 = await sendEmailContext.newPage();
+        await page2.setCookie(...cookies);
+
+        await page2.goto('https://mail.yahoo.com/');
+
+        await page2.click('a[aria-label="Compose"]');
+
+        await page2.type('input#message-to-field', to);
+        await page2.type('input[data-test-id="compose-subject"]', subject);
+        await page2.type('div[data-test-id="compose-editor-container"] div[contenteditable="true"]', body);
+
+        await page2.click('button[title="Send this email"]');
+
+        await sendEmailContext.close();
+    })()
+);
+
+await Promise.all(promises);
+
+await browser.close();
+</marked-tab>
+```
+
```

Comment:
1. Not sure what you agreed with this lesson but I would also add Crawlee code. That is what 90% of our users will be using so they need to be able to almost copy/paste that.
2. Mention that we can also pass in cookies to the actor and only inject them
(if you think these do not belong here, we should add them elsewhere)

Reply:
Screenshotting your comment + adding it to backlog.